jpegenc/subblocks/dct_1d/dct_1d.py

```diff
@@ -128,23 +128,23 @@ def coeff_assign():
     def mul_add():
         """multiplication and addition"""
         if input_interface.data_valid:
-                mult_reg[0].next = data_in_reg * coeffs[0]
-                mult_reg[1].next = data_in_reg * coeffs[1]
-                mult_reg[2].next = data_in_reg * coeffs[2]
-                mult_reg[3].next = data_in_reg * coeffs[3]
-                mult_reg[4].next = data_in_reg * coeffs[4]
-                mult_reg[5].next = data_in_reg * coeffs[5]
-                mult_reg[6].next = data_in_reg * coeffs[6]
-                mult_reg[7].next = data_in_reg * coeffs[7]
-
-                adder_reg[0].next = mux_flush[0] + mult_reg[0]
-                adder_reg[1].next = mux_flush[1] + mult_reg[1]
-                adder_reg[2].next = mux_flush[2] + mult_reg[2]
-                adder_reg[3].next = mux_flush[3] + mult_reg[3]
-                adder_reg[4].next = mux_flush[4] + mult_reg[4]
-                adder_reg[5].next = mux_flush[5] + mult_reg[5]
-                adder_reg[6].next = mux_flush[6] + mult_reg[6]
-                adder_reg[7].next = mux_flush[7] + mult_reg[7]
+            mult_reg[0].next = data_in_reg * coeffs[0]
```

Comment:
I don't recall the being the situations (I think I have some counter examples) but I will need to search for them.

If device specific optimization are implemented these need to be called out specifically (comments, docstrings, etc.).  Personally, I don't like premature optimization, I would rather see this a clean and a modular as possible, get some data resource usage, and then iterate (figure out how to include the optimizations.